static/js/services/message-contacts.js

```diff
@@ -12,7 +12,7 @@ var async = require('async'),
     function(HttpWrapper, BaseUrlService) {
       return function(params, callback) {
         var url = BaseUrlService() + '/message_contacts';
-        HttpWrapper.get(url, { params: params })
+        HttpWrapper.get(url, { params: params, timeout: false })
```

Comment:
Another possible solution - the `{ timeout: false }` option is used mainly in the root scope because you can't navigate away from the root scope and still be in the app. I think #925 occurs when the user changes from route `messages.a` to `messages.b`. In this case you haven't navigated away from the parent `messages` route, but you would want to cancel all requests that would affect the view of the child `messages.a` route.

In essence you only want to cancel requests made by a route you're navigating away from - if you still within Messages but have changed the selected Message then anything that would update the LHS should be allowed to continue, but anything that would update the RHS should be cancelled. Likewise if you're navigating to Reports anything that updates the overall Inbox should be allowed to continue but anything that updates the Messages LHS or RHS should be cancelled.

So the more correct solution would be for every call to the HttpWrapper to provide the route which will be updated. Then when a route change event occurs pass in the new route ( https://github.com/angular-ui/ui-router/wiki#state-change-events ), and only cancel requests that aren't part of the new route.

For bonus points instead of passing in the route, HttpWrapper can query $state.$current to automatically determine the correct state.

This is quite complex but will probably solve this issue and a bunch of other ones we haven't found yet. It also removes the hacky `timeout: false` option.